peripherals/adc/adc_f1.yaml

```diff
@@ -0,0 +1,156 @@
+      Enabled: [1, "Conversion on external event enabled"]
+    JEXTTRIG:
+      Disabled: [0, "Conversion on external event disabled"]
+      Enabled: [1, "Conversion on external event enabled"]
+    ALIGN:
+      Right: [0, "Right Alignment"]
+      Left: [1, "Left Alignment"]
+    RSTCAL:
+      _read:
+        Initialized: [0, "Calibration register initialized"]
+        NotInitialized: [1, "Initializing calibration register"]
+      _write:
+        Initialize: [1, "Initialize calibration register"]
+    CAL:
+      _read:
+        Complete: [0, "Calibration completed"]
+        NotComplete: [1, "Calibrating"]
+      _write:
+        Start: [1, "Enable calibration"]
+    CONT:
+      Single: [0, "Single conversion mode"]
+      Continious: [1, "Continuous conversion mode"]
+    ADON:
+      Disable: [0, "Disable ADC conversion/calibration and go to power down mode"]
+      Enable: [1, "Enable ADC and to start conversion"]
+
+  "SMPR?":
+    "SMP*":
+      Cycles1_5: [0, "1.5 ADC clock cycles"]
+      Cycles7_5: [1, "7.5 ADC clock cycles"]
+      Cycles13_5: [2, "13.5 ADC clock cycles"]
+      Cycles28_5: [3, "28.5 ADC clock cycles"]
+      Cycles41_5: [4, "41.5 ADC clock cycles"]
+      Cycles55_5: [5, "55.5 ADC clock cycles"]
+      Cycles71_5: [6, "71.5 ADC clock cycles"]
+      Cycles239_5: [7, "239.5 ADC clock cycles"]
+  "SQR?":
+    "SQ*": [0, 17]
+
+ADC1:
+  CR1:
+    DUALMOD:
+      Independent: [0, "Independent mode"]
+      RegularInjected: [1, "Combined regular simultaneous + injected simultaneous mode"]
+      RegularAlternateTrigger: [2, "Combined regular simultaneous + alternate trigger mode"]
+      InjectedFastInterleaved: [3, "Combined injected simultaneous + fast interleaved mode"]
+      InjectedSlowInterleaved: [4, "Combined injected simultaneous + slow interleaved mode"]
+      Injected: [5, "Injected simultaneous mode only"]
+      Regular: [6, "Regular simultaneous mode only"]
+      FastInterleaved: [7, "Fast interleaved mode only"]
+      SlowInterleaved: [8, "Slow interleaved mode only"]
+      AlternateTrigger: [9, "Alternate trigger mode only"]
+  CR2:
+    EXTSEL:
+      Tim1Cc1: [0, "Timer 1 CC1 event"]
+      Tim1Cc2: [1, "Timer 1 CC2 event"]
+      Tim1Cc3: [2, "Timer 1 CC3 event"]
+      Tim2Cc2: [3, "Timer 2 CC2 event"]
+      Tim3Trgo: [4, "Timer 3 TRGO event"]
+      Tim4Cc4: [5, "Timer 4 CC4 event"]
+      Exti11: [6, "EXTI line 11/TIM8_TRGO event (TIM8_TRGO is available only in high-density and XL-density devices)"]
+      Swstart: [7, "SWSTART"]
+    JEXTSEL:
+      Tim1Trgo: [0, "Timer 1 TRGO event"]
+      Tim1Cc4: [1, "Timer 1 CC4 event"]
+      Tim2Trgo: [2, "Timer 2 TRGO event"]
+      Tim2Cc1: [3, "Timer 2 CC1 event"]
+      Tim3Cc4: [4, "Timer 3 CC4 event"]
+      Tim4Trgo: [5, "Timer 4 TRGO event"]
+      Exti15: [6, "EXTI line15/TIM8_CC4 event (TIM8_CC4 is available only in high-density and XL-density devices)"]
+      Jswstart: [7, "JSWSTART"]
```

Comment:
`DMA` field is missing but I think could be specified here (it's applicable for ADC1 and ADC3, but same problem with ADC2 as mentioned above, so at least adding it for ADC1 would be useful).